### ISSUE TITLE
A11Y error with keyboard navigation on gallery when using full screen carousel

### ISSUE THREAD

**Issue by NicktheGeek** [Automattic/jetpack#28291]
### Impacted plugin

Jetpack

### Steps to Reproduce

1. Enable the "Display images in a full screen carousel gallery" feature
2. Add a gallery to a post or page
3. Verify that clicking on the gallery items opens the carousel
4. Close the carousel and attempt to navigate the page with a keyboard
5. Note it is impossible to trigger the carousel because images are not focusable
6. Turn on a screen reader and attempt to navigate the page
7. Note it is impossible to trigger the carousel because images are not focusable

### A clear and concise description of what you expected to happen.

At the minimum tabindex=0 should be added to the images and they should get role=button.

This may also benefit from additional aria attributes such as `aria-controls` and `aria-expanded`. 

The JS trigger may need to include an event listener for space and enter if that doesn't already exist. Since this is a "button" it must respond to space, trigger, click, and tap (touch) events.

### What actually happened

It is impossible to trigger the carousel because images are not focusable

This is a failure of WCAG 2.1.1 Keyboard.

This is a level A success criterion which means this is a critical issue as sites cannot be considered accessible if this feature is enabled. 

### Browser

Google Chrome/Chromium, Mozilla Firefox, Microsoft Edge, Apple Safari

### Reproducibility

Consistent

### Severity

None

### Available workarounds?

None

**Comment by tabbplomaritas** [comment]
This is still a problem. This is a pretty major accessibility issue and it's been a year since this issue was reported. Can we get this fixed?
